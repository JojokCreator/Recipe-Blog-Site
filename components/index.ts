export { default as PostCard} from './PostCard';
export { default as Categories} from './Categories';
export { default as PostWidget} from './PostWidget';
export { default as Header} from './Header';
export { default as Author} from './Author';
export { default as CommentForm} from './CommentForm';
export { default as Comments} from './Comments';
export { default as PostDetail} from './PostDetail';
export { default as FeaturedPostCard} from './FeaturedPostCard';
export { default as AdjacentPostCard} from './AdjacentPostCard';
export { default as Loader} from './Loader';
export { default as Footer} from './Footer';