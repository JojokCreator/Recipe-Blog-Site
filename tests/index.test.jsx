import { act, fireEvent, render, screen } from '@testing-library/react'
import { getCategories, getPosts } from '../services'
import { PostCard, Home, Header } from '../components'

describe('Home', () => {
  it('renders a heading', async () => {
    const categories = await getCategories()
    render(<Header categories={categories} />)
    const heading = screen.getByText('Barefoot Chef Blog')
    expect(heading).toBeInTheDocument()
  })

  it('snapshot matches previous', async () => {
    const posts = await getPosts()
    const { container } = render(<Home posts={posts} />)
    expect(container).toMatchSnapshot()
  })

  it('renders initial 6 posts', async () => {
    const posts = await getPosts()
    render(<Home posts={posts} />)
    const heading = screen.getAllByRole('listitem')
    expect(heading.length).toBe(6)
  })
})

describe('PostCard', () => {
  it('renders a heading', async () => {
    const posts = await getPosts()

    render(<PostCard post={posts[0].node} />)

    const heading = screen.getByText('Pork Adobo')

    expect(heading).toBeInTheDocument()
  })

  it('snapshot matches previous', async () => {
    const posts = await getPosts()
    const { container } = render(<PostCard post={posts[0].node} />)

    expect(container).toMatchSnapshot()
  })

  it('renders initial 6 posts', async () => {
    const posts = await getPosts()
    render(<Home posts={posts} />)
    const heading = screen.getAllByRole('link')
    expect(heading.length).toBe(12)
  })

  describe('Search Bar', () => {
    it('shows no posts found if search is xxx445', async () => {
      const posts = await getPosts()
      render(<Home posts={posts} />)
      const input = screen.getByPlaceholderText('Search Here...')
      const button = screen.getByText('Search')
      fireEvent.change(input, { target: { value: 'xxx445' } })
      fireEvent.click(button)

      const heading = screen.getByText('No posts found')
      expect(heading).toBeInTheDocument()
    })
    it('shows one posts found if search is pork', async () => {
      const posts = await getPosts()
      render(<Home posts={posts} />)
      const input = screen.getByPlaceholderText('Search Here...')
      const button = screen.getByText('Search')
      fireEvent.change(input, { target: { value: 'pork' } })
      fireEvent.click(button)

      const heading = screen.getAllByRole('listitem')
      expect(heading.length).toBe(1)
      screen.debug(heading)
    })
  })
})
