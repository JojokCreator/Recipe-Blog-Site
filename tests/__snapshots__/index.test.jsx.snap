// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home snapshot matches previous 1`] = `
<div>
  <div
    class="container mx-auto px-10 mb-8"
  >
    <div
      class="mb-8"
    >
      <div
        class="react-multi-carousel-list  "
        dir="ltr"
      >
        <ul
          class="react-multi-carousel-track "
          style="transition: none; overflow: unset; transform: translate3d(0px,0,0);"
        />
      </div>
    </div>
    <div
      class="flex p-4 flex-col md:flex-row"
    >
      <input
        class="py-2 px-4 w-full ring-black rounded-lg focus:ring-2 focus:ring-gray-200 bg-gray-10 text-center mb-4"
        name="search"
        placeholder="Search Here..."
        type="text"
        value=""
      />
      <button
        aria-label="Click to search the posts"
        class="transition duration-500 ease hover:bg-indigo-900 inline-block bg-indigo-600 text-lg font-medium rounded-full text-white px-8 py-3 cursor-pointer ml-2"
        type="button"
      >
        Search
      </button>
    </div>
    <div
      class="grid grid-cols-1 lg:grid-cols-12 gap-12"
    >
      <ul
        class="lg:col-span-8 col-span-1 list-none"
      >
        <li
          class="bg-white shadow-lg rounded-lg p-2 lg:p-8 pb-12 mb-8"
        >
          <div
            class="relative overflow-hidden shadow-md pb-80 mb-6"
          >
            <img
              alt="Pork Adobo"
              class="object-top absolute h-80 w-full object-cover shadow-lg rounded-t-lg lg:rounded-lg"
              data-nimg="fill"
              decoding="async"
              loading="lazy"
              sizes="100vw"
              src="/_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FKjc23uhUSla5Fy4Y0311&w=3840&q=75"
              srcset="/_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FKjc23uhUSla5Fy4Y0311&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FKjc23uhUSla5Fy4Y0311&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FKjc23uhUSla5Fy4Y0311&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FKjc23uhUSla5Fy4Y0311&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FKjc23uhUSla5Fy4Y0311&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FKjc23uhUSla5Fy4Y0311&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FKjc23uhUSla5Fy4Y0311&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FKjc23uhUSla5Fy4Y0311&w=3840&q=75 3840w"
              style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
            />
          </div>
          <h1
            class="transition duration-700 text-center mb-8 cursor-pointer hover:text-indigo-600 text-3xl font-semibold "
          >
            <a
              href="/post/pork-adobo"
            >
              Pork Adobo
            </a>
          </h1>
          <div
            class="block lg:flex text-center items-center justify-center mb-8 w-full"
          >
            <div
              class="flex items-cent justify-center mb-4 lg:mb-0 w-full lg:w-auto mr-8"
            >
              <img
                alt="Barefoot Chef"
                class="align-middle rounded-full"
                data-nimg="1"
                decoding="async"
                height="30"
                loading="lazy"
                src="/_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FexE6aAjcQ06f2PXndvnS&w=64&q=75"
                srcset="/_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FexE6aAjcQ06f2PXndvnS&w=32&q=75 1x, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FexE6aAjcQ06f2PXndvnS&w=64&q=75 2x"
                style="color: transparent;"
                width="30"
              />
              <p
                class="inline align-middle text-grey-700 ml-2 text-lg"
              >
                Barefoot Chef
              </p>
            </div>
            <div
              class="font-medium text-gray-700"
            >
              <span>
                Jan 07, 2023
              </span>
            </div>
          </div>
          <p
            class="text-center text-lg text-grey-700 font-normal px-4 lg:px-20 mb-8"
          >
               Adobo is a popular dish in the Philippines that is made by braising pork in a mixture of soy sauce, vinegar, and spices.
          </p>
          <div
            class="text-center"
            role="button"
          >
            <a
              aria-label="Read more about Pork Adobo"
              href="/post/pork-adobo"
            >
              <span
                class="transition duration-500 transform hover:-translate-y-1 inline-block bg-indigo-600 text-lg font-medium rounded-full text-white px-8 py-3 cursor-pointer"
              >
                Continue Reading
              </span>
            </a>
          </div>
        </li>
        <li
          class="bg-white shadow-lg rounded-lg p-2 lg:p-8 pb-12 mb-8"
        >
          <div
            class="relative overflow-hidden shadow-md pb-80 mb-6"
          >
            <img
              alt="Rendang - Slow cooked Dry Curry"
              class="object-top absolute h-80 w-full object-cover shadow-lg rounded-t-lg lg:rounded-lg"
              data-nimg="fill"
              decoding="async"
              loading="lazy"
              sizes="100vw"
              src="/_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FVDx0OccQDCQO2QkbboK2&w=3840&q=75"
              srcset="/_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FVDx0OccQDCQO2QkbboK2&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FVDx0OccQDCQO2QkbboK2&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FVDx0OccQDCQO2QkbboK2&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FVDx0OccQDCQO2QkbboK2&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FVDx0OccQDCQO2QkbboK2&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FVDx0OccQDCQO2QkbboK2&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FVDx0OccQDCQO2QkbboK2&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FVDx0OccQDCQO2QkbboK2&w=3840&q=75 3840w"
              style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
            />
          </div>
          <h1
            class="transition duration-700 text-center mb-8 cursor-pointer hover:text-indigo-600 text-3xl font-semibold "
          >
            <a
              href="/post/rendang"
            >
              Rendang - Slow cooked Dry Curry
            </a>
          </h1>
          <div
            class="block lg:flex text-center items-center justify-center mb-8 w-full"
          >
            <div
              class="flex items-cent justify-center mb-4 lg:mb-0 w-full lg:w-auto mr-8"
            >
              <img
                alt="Barefoot Chef"
                class="align-middle rounded-full"
                data-nimg="1"
                decoding="async"
                height="30"
                loading="lazy"
                src="/_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FexE6aAjcQ06f2PXndvnS&w=64&q=75"
                srcset="/_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FexE6aAjcQ06f2PXndvnS&w=32&q=75 1x, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FexE6aAjcQ06f2PXndvnS&w=64&q=75 2x"
                style="color: transparent;"
                width="30"
              />
              <p
                class="inline align-middle text-grey-700 ml-2 text-lg"
              >
                Barefoot Chef
              </p>
            </div>
            <div
              class="font-medium text-gray-700"
            >
              <span>
                Dec 26, 2022
              </span>
            </div>
          </div>
          <p
            class="text-center text-lg text-grey-700 font-normal px-4 lg:px-20 mb-8"
          >
            Rendang is a dry curry which includes a mixture of spices and coconut milk, and then slow-cooked over low heat until it is tender and falling apart. The sauce is reduced and thickened as the meat cooks, creating a rich and flavorful coating that is perfect for serving over rice.
          </p>
          <div
            class="text-center"
            role="button"
          >
            <a
              aria-label="Read more about Rendang - Slow cooked Dry Curry"
              href="/post/rendang"
            >
              <span
                class="transition duration-500 transform hover:-translate-y-1 inline-block bg-indigo-600 text-lg font-medium rounded-full text-white px-8 py-3 cursor-pointer"
              >
                Continue Reading
              </span>
            </a>
          </div>
        </li>
        <li
          class="bg-white shadow-lg rounded-lg p-2 lg:p-8 pb-12 mb-8"
        >
          <div
            class="relative overflow-hidden shadow-md pb-80 mb-6"
          >
            <img
              alt="Laksa - Curry Prawn Noodle Soup"
              class="object-top absolute h-80 w-full object-cover shadow-lg rounded-t-lg lg:rounded-lg"
              data-nimg="fill"
              decoding="async"
              loading="lazy"
              sizes="100vw"
              src="/_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2F79xzZtbvRLKEEtbbWzsa&w=3840&q=75"
              srcset="/_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2F79xzZtbvRLKEEtbbWzsa&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2F79xzZtbvRLKEEtbbWzsa&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2F79xzZtbvRLKEEtbbWzsa&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2F79xzZtbvRLKEEtbbWzsa&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2F79xzZtbvRLKEEtbbWzsa&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2F79xzZtbvRLKEEtbbWzsa&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2F79xzZtbvRLKEEtbbWzsa&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2F79xzZtbvRLKEEtbbWzsa&w=3840&q=75 3840w"
              style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
            />
          </div>
          <h1
            class="transition duration-700 text-center mb-8 cursor-pointer hover:text-indigo-600 text-3xl font-semibold "
          >
            <a
              href="/post/laksa-prawn-curry"
            >
              Laksa - Curry Prawn Noodle Soup
            </a>
          </h1>
          <div
            class="block lg:flex text-center items-center justify-center mb-8 w-full"
          >
            <div
              class="flex items-cent justify-center mb-4 lg:mb-0 w-full lg:w-auto mr-8"
            >
              <img
                alt="Barefoot Chef"
                class="align-middle rounded-full"
                data-nimg="1"
                decoding="async"
                height="30"
                loading="lazy"
                src="/_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FexE6aAjcQ06f2PXndvnS&w=64&q=75"
                srcset="/_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FexE6aAjcQ06f2PXndvnS&w=32&q=75 1x, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FexE6aAjcQ06f2PXndvnS&w=64&q=75 2x"
                style="color: transparent;"
                width="30"
              />
              <p
                class="inline align-middle text-grey-700 ml-2 text-lg"
              >
                Barefoot Chef
              </p>
            </div>
            <div
              class="font-medium text-gray-700"
            >
              <span>
                Sep 13, 2022
              </span>
            </div>
          </div>
          <p
            class="text-center text-lg text-grey-700 font-normal px-4 lg:px-20 mb-8"
          >
            There are about as many different variations of Laksa as there are towns and city's in Malaysia. Some add more or less spice, clams, more or less coconut. The noodles also vary some people prefer rice noodles but for me the egg noodles are the best.
          </p>
          <div
            class="text-center"
            role="button"
          >
            <a
              aria-label="Read more about Laksa - Curry Prawn Noodle Soup"
              href="/post/laksa-prawn-curry"
            >
              <span
                class="transition duration-500 transform hover:-translate-y-1 inline-block bg-indigo-600 text-lg font-medium rounded-full text-white px-8 py-3 cursor-pointer"
              >
                Continue Reading
              </span>
            </a>
          </div>
        </li>
        <li
          class="bg-white shadow-lg rounded-lg p-2 lg:p-8 pb-12 mb-8"
        >
          <div
            class="relative overflow-hidden shadow-md pb-80 mb-6"
          >
            <img
              alt="Homemade Rice Wine (Choujiu)"
              class="object-top absolute h-80 w-full object-cover shadow-lg rounded-t-lg lg:rounded-lg"
              data-nimg="fill"
              decoding="async"
              loading="lazy"
              sizes="100vw"
              src="/_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FpgfBIlNUTKIN4IOPTcgs&w=3840&q=75"
              srcset="/_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FpgfBIlNUTKIN4IOPTcgs&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FpgfBIlNUTKIN4IOPTcgs&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FpgfBIlNUTKIN4IOPTcgs&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FpgfBIlNUTKIN4IOPTcgs&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FpgfBIlNUTKIN4IOPTcgs&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FpgfBIlNUTKIN4IOPTcgs&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FpgfBIlNUTKIN4IOPTcgs&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FpgfBIlNUTKIN4IOPTcgs&w=3840&q=75 3840w"
              style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
            />
          </div>
          <h1
            class="transition duration-700 text-center mb-8 cursor-pointer hover:text-indigo-600 text-3xl font-semibold "
          >
            <a
              href="/post/rice-wine"
            >
              Homemade Rice Wine (Choujiu)
            </a>
          </h1>
          <div
            class="block lg:flex text-center items-center justify-center mb-8 w-full"
          >
            <div
              class="flex items-cent justify-center mb-4 lg:mb-0 w-full lg:w-auto mr-8"
            >
              <img
                alt="Barefoot Chef"
                class="align-middle rounded-full"
                data-nimg="1"
                decoding="async"
                height="30"
                loading="lazy"
                src="/_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FexE6aAjcQ06f2PXndvnS&w=64&q=75"
                srcset="/_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FexE6aAjcQ06f2PXndvnS&w=32&q=75 1x, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FexE6aAjcQ06f2PXndvnS&w=64&q=75 2x"
                style="color: transparent;"
                width="30"
              />
              <p
                class="inline align-middle text-grey-700 ml-2 text-lg"
              >
                Barefoot Chef
              </p>
            </div>
            <div
              class="font-medium text-gray-700"
            >
              <span>
                Sep 13, 2022
              </span>
            </div>
          </div>
          <p
            class="text-center text-lg text-grey-700 font-normal px-4 lg:px-20 mb-8"
          >
            This Choujiu style rice wine is made with glutinous rice and fermented by a very specific yeast. It's similar to Makgeolli in Korea or Doburoku in Japan. It's generally drunk young as it has been for thousands of years.
          </p>
          <div
            class="text-center"
            role="button"
          >
            <a
              aria-label="Read more about Homemade Rice Wine (Choujiu)"
              href="/post/rice-wine"
            >
              <span
                class="transition duration-500 transform hover:-translate-y-1 inline-block bg-indigo-600 text-lg font-medium rounded-full text-white px-8 py-3 cursor-pointer"
              >
                Continue Reading
              </span>
            </a>
          </div>
        </li>
        <li
          class="bg-white shadow-lg rounded-lg p-2 lg:p-8 pb-12 mb-8"
        >
          <div
            class="relative overflow-hidden shadow-md pb-80 mb-6"
          >
            <img
              alt="Red Rice Wine Chicken – 红槽鸡"
              class="object-top absolute h-80 w-full object-cover shadow-lg rounded-t-lg lg:rounded-lg"
              data-nimg="fill"
              decoding="async"
              loading="lazy"
              sizes="100vw"
              src="/_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FZhfSPVDWTrC37nVHNrPt&w=3840&q=75"
              srcset="/_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FZhfSPVDWTrC37nVHNrPt&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FZhfSPVDWTrC37nVHNrPt&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FZhfSPVDWTrC37nVHNrPt&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FZhfSPVDWTrC37nVHNrPt&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FZhfSPVDWTrC37nVHNrPt&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FZhfSPVDWTrC37nVHNrPt&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FZhfSPVDWTrC37nVHNrPt&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FZhfSPVDWTrC37nVHNrPt&w=3840&q=75 3840w"
              style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
            />
          </div>
          <h1
            class="transition duration-700 text-center mb-8 cursor-pointer hover:text-indigo-600 text-3xl font-semibold "
          >
            <a
              href="/post/red-rice-chicken-noodles"
            >
              Red Rice Wine Chicken – 红槽鸡
            </a>
          </h1>
          <div
            class="block lg:flex text-center items-center justify-center mb-8 w-full"
          >
            <div
              class="flex items-cent justify-center mb-4 lg:mb-0 w-full lg:w-auto mr-8"
            >
              <img
                alt="Barefoot Chef"
                class="align-middle rounded-full"
                data-nimg="1"
                decoding="async"
                height="30"
                loading="lazy"
                src="/_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FexE6aAjcQ06f2PXndvnS&w=64&q=75"
                srcset="/_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FexE6aAjcQ06f2PXndvnS&w=32&q=75 1x, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FexE6aAjcQ06f2PXndvnS&w=64&q=75 2x"
                style="color: transparent;"
                width="30"
              />
              <p
                class="inline align-middle text-grey-700 ml-2 text-lg"
              >
                Barefoot Chef
              </p>
            </div>
            <div
              class="font-medium text-gray-700"
            >
              <span>
                Sep 13, 2022
              </span>
            </div>
          </div>
          <p
            class="text-center text-lg text-grey-700 font-normal px-4 lg:px-20 mb-8"
          >
            This typical Foochow dish called hong jao zi. The rich, red coloured dish is flavoured with thick, aromatic fermented red wine. This dish is traditionally made for mothers during the confinement period after childbirth. Healthy and nutritious enjoyed anytime though.
          </p>
          <div
            class="text-center"
            role="button"
          >
            <a
              aria-label="Read more about Red Rice Wine Chicken – 红槽鸡"
              href="/post/red-rice-chicken-noodles"
            >
              <span
                class="transition duration-500 transform hover:-translate-y-1 inline-block bg-indigo-600 text-lg font-medium rounded-full text-white px-8 py-3 cursor-pointer"
              >
                Continue Reading
              </span>
            </a>
          </div>
        </li>
        <li
          class="bg-white shadow-lg rounded-lg p-2 lg:p-8 pb-12 mb-8"
        >
          <div
            class="relative overflow-hidden shadow-md pb-80 mb-6"
          >
            <img
              alt="Red Yeast Rice Wine (Fuzhou Hong Zhao)"
              class="object-top absolute h-80 w-full object-cover shadow-lg rounded-t-lg lg:rounded-lg"
              data-nimg="fill"
              decoding="async"
              loading="lazy"
              sizes="100vw"
              src="/_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2Fxzzp4CCGTiWotCiENikC&w=3840&q=75"
              srcset="/_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2Fxzzp4CCGTiWotCiENikC&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2Fxzzp4CCGTiWotCiENikC&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2Fxzzp4CCGTiWotCiENikC&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2Fxzzp4CCGTiWotCiENikC&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2Fxzzp4CCGTiWotCiENikC&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2Fxzzp4CCGTiWotCiENikC&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2Fxzzp4CCGTiWotCiENikC&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2Fxzzp4CCGTiWotCiENikC&w=3840&q=75 3840w"
              style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
            />
          </div>
          <h1
            class="transition duration-700 text-center mb-8 cursor-pointer hover:text-indigo-600 text-3xl font-semibold "
          >
            <a
              href="/post/red-yeast-rice-wine"
            >
              Red Yeast Rice Wine (Fuzhou Hong Zhao)
            </a>
          </h1>
          <div
            class="block lg:flex text-center items-center justify-center mb-8 w-full"
          >
            <div
              class="flex items-cent justify-center mb-4 lg:mb-0 w-full lg:w-auto mr-8"
            >
              <img
                alt="Barefoot Chef"
                class="align-middle rounded-full"
                data-nimg="1"
                decoding="async"
                height="30"
                loading="lazy"
                src="/_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FexE6aAjcQ06f2PXndvnS&w=64&q=75"
                srcset="/_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FexE6aAjcQ06f2PXndvnS&w=32&q=75 1x, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FexE6aAjcQ06f2PXndvnS&w=64&q=75 2x"
                style="color: transparent;"
                width="30"
              />
              <p
                class="inline align-middle text-grey-700 ml-2 text-lg"
              >
                Barefoot Chef
              </p>
            </div>
            <div
              class="font-medium text-gray-700"
            >
              <span>
                Sep 13, 2022
              </span>
            </div>
          </div>
          <p
            class="text-center text-lg text-grey-700 font-normal px-4 lg:px-20 mb-8"
          >
            Used for as an ingriedient than a drink red rice wine can be used as a seasoning for basically anything for example, chicken, pork, fish, tofu, rice, noodles. Fermentation is carried out by a combination of the fungus Aspergillus oryzae, which converts the rice starches into fermentable sugars, and yeast which inturns converts the sugars into alcohol.
          </p>
          <div
            class="text-center"
            role="button"
          >
            <a
              aria-label="Read more about Red Yeast Rice Wine (Fuzhou Hong Zhao)"
              href="/post/red-yeast-rice-wine"
            >
              <span
                class="transition duration-500 transform hover:-translate-y-1 inline-block bg-indigo-600 text-lg font-medium rounded-full text-white px-8 py-3 cursor-pointer"
              >
                Continue Reading
              </span>
            </a>
          </div>
        </li>
      </ul>
      <div
        class="lg:col-span-4 cols-span-1"
      >
        <div
          class="lg:sticky relative top-8"
        >
          <div
            class="bg-white shadow-lg rounded-lg p-8 mb-8"
          >
            <h3
              class="text-xl mb-8 font-semibold border-b pb-4"
            >
              Recent Posts
            </h3>
          </div>
          <div
            class="bg-white shadow-lg rounded-lg p-8 mb-8"
          >
            <h3
              class="text-xl mb-8 font-semibold border-b pb-4"
            >
              Categories
            </h3>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`PostCard snapshot matches previous 1`] = `
<div>
  <li
    class="bg-white shadow-lg rounded-lg p-2 lg:p-8 pb-12 mb-8"
  >
    <div
      class="relative overflow-hidden shadow-md pb-80 mb-6"
    >
      <img
        alt="Pork Adobo"
        class="object-top absolute h-80 w-full object-cover shadow-lg rounded-t-lg lg:rounded-lg"
        data-nimg="fill"
        decoding="async"
        loading="lazy"
        sizes="100vw"
        src="/_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FKjc23uhUSla5Fy4Y0311&w=3840&q=75"
        srcset="/_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FKjc23uhUSla5Fy4Y0311&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FKjc23uhUSla5Fy4Y0311&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FKjc23uhUSla5Fy4Y0311&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FKjc23uhUSla5Fy4Y0311&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FKjc23uhUSla5Fy4Y0311&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FKjc23uhUSla5Fy4Y0311&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FKjc23uhUSla5Fy4Y0311&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FKjc23uhUSla5Fy4Y0311&w=3840&q=75 3840w"
        style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
      />
    </div>
    <h1
      class="transition duration-700 text-center mb-8 cursor-pointer hover:text-indigo-600 text-3xl font-semibold "
    >
      <a
        href="/post/pork-adobo"
      >
        Pork Adobo
      </a>
    </h1>
    <div
      class="block lg:flex text-center items-center justify-center mb-8 w-full"
    >
      <div
        class="flex items-cent justify-center mb-4 lg:mb-0 w-full lg:w-auto mr-8"
      >
        <img
          alt="Barefoot Chef"
          class="align-middle rounded-full"
          data-nimg="1"
          decoding="async"
          height="30"
          loading="lazy"
          src="/_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FexE6aAjcQ06f2PXndvnS&w=64&q=75"
          srcset="/_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FexE6aAjcQ06f2PXndvnS&w=32&q=75 1x, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FexE6aAjcQ06f2PXndvnS&w=64&q=75 2x"
          style="color: transparent;"
          width="30"
        />
        <p
          class="inline align-middle text-grey-700 ml-2 text-lg"
        >
          Barefoot Chef
        </p>
      </div>
      <div
        class="font-medium text-gray-700"
      >
        <span>
          Jan 07, 2023
        </span>
      </div>
    </div>
    <p
      class="text-center text-lg text-grey-700 font-normal px-4 lg:px-20 mb-8"
    >
         Adobo is a popular dish in the Philippines that is made by braising pork in a mixture of soy sauce, vinegar, and spices.
    </p>
    <div
      class="text-center"
      role="button"
    >
      <a
        aria-label="Read more about Pork Adobo"
        href="/post/pork-adobo"
      >
        <span
          class="transition duration-500 transform hover:-translate-y-1 inline-block bg-indigo-600 text-lg font-medium rounded-full text-white px-8 py-3 cursor-pointer"
        >
          Continue Reading
        </span>
      </a>
    </div>
  </li>
</div>
`;
